### PR TITLE
BIZ-1 Main infrastructure - Server

### PR DESCRIPTION
**Pull request by RomanFedytskyi**

### CODE REVIEW

File: app/server.js [hunk 1]
```diff
@@ -52,6 +54,19 @@ module.exports = function () {
     routing.init(server);
   }
 
+  function registerACL(server) {
```
Comment (dmitrij-borchuk) [inline app/server.js]: Where do you call it?

File: app/server.js [hunk 2]
```diff
@@ -52,6 +54,19 @@ module.exports = function () {
...
+    return new Promise(function (resolve, reject) {
+      require('./acl.js')(server, function(err) {
+        if (err) {
+          console.log(err);
```
Comment (dmitrij-borchuk) [inline app/server.js]: Remove console.log

File: app/server.js [hunk 3]
```diff
@@ -52,6 +54,19 @@ module.exports = function () {
     routing.init(server);
   }
 
+  function registerACL(server) {
+    return new Promise(function (resolve, reject) {
+      require('./acl.js')(server, function(err) {
+        if (err) {
```
Comment (dmitrij-borchuk) [inline app/server.js]: Use short form:
err ? reject() : resolve();

File: app/dataConnection.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+var mysql = require('mysql');
```
Comment (dmitrij-borchuk) [inline app/dataConnection.js]: Where do you use this file?

File: app/server.js [hunk 5]
```diff
@@ -66,4 +81,34 @@ module.exports = function () {
       }
     );
   }
+
+  function registerAuth(server) {
+    return new Promise(function (resolve, reject) {
+       const AuthHeader = require('hapi-auth-header');
```
Comment (dmitrij-borchuk) [inline app/server.js]: Fix identation

File: app/server.js [hunk 6]
```diff
@@ -66,4 +81,34 @@ module.exports = function () {
...
+
+      server.register(AuthHeader, (err) => {
+        if (err) {
+          reject();
+        } else {
+           server.auth.strategy('simple', 'auth-header', {
```
Comment (dmitrij-borchuk) [inline app/server.js]: Fix identation

File: app/server.js [hunk 7]
```diff
@@ -66,4 +81,34 @@ module.exports = function () {
...
+            accessTokenName: 'X-CART-Token',
+            validateFunc: function (tokens, callback) {
```
Comment (dmitrij-borchuk) [inline app/server.js]: Move this function to end if file

File: app/server.js [hunk 8]
```diff
@@ -66,4 +81,34 @@ module.exports = function () {
...
+              var request = this;
```
Comment (dmitrij-borchuk) [inline app/server.js]: Where do you use this variable?

File: app/server.js [hunk 9]
```diff
@@ -66,4 +81,34 @@ module.exports = function () {
...
+              var headerName = 'X-CART-Token';
+
+               DAL.users.getUserByToken(tokens[headerName], function (err, user) {
```
Comment (dmitrij-borchuk) [inline app/server.js]: Dal is not exist

**Review by dmitrij-borchuk — CHANGES_REQUESTED** [review]

File: app/server.js (hunk 8 above)
Comment (RomanFedytskyi) [reply app/server.js]: fixed

File: app/server.js (hunk 1 above)
Comment (RomanFedytskyi) [reply app/server.js]: fixed

File: app/server.js (hunk 3 above)
Comment (RomanFedytskyi) [reply app/server.js]: fixed

File: app/server.js (hunk 2 above)
Comment (RomanFedytskyi) [reply app/server.js]: fixed

File: app/server.js (hunk 9 above)
Comment (RomanFedytskyi) [reply app/server.js]: fixed

File: app/dataConnection.js (hunk 4 above)
Comment (RomanFedytskyi) [reply app/dataConnection.js]: var connection = mysql.createConnection({

File: backend/server.js [hunk 10]
```diff
@@ -49,11 +50,28 @@ module.exports = function () {
     );
   }
 
+  function connectToDB() {
```
Comment (dmitrij-borchuk) [inline backend/server.js]: Where do you call it?

File: backend/dataConnection.js [hunk 11]
```diff
@@ -1,24 +1,25 @@
...
+    }
+  });
 
-connection.query('SELECT * FROM pvmDB', function(err, rows, fields) {
-  if (!err)
-    console.log('The solution is: ', rows);
-  else
-    console.log('Error while performing Query.');
-});
+  connection.query('SELECT * FROM pvmDB', function(err, rows, fields) {
```
Comment (dmitrij-borchuk) [inline backend/dataConnection.js]: What for?

File: backend/server.js [hunk 12]
```diff
@@ -49,11 +50,28 @@ module.exports = function () {
...
+
   function registerRouting(server) {
     const routing = require('./routing');
     routing.init(server);
   }
 
+  function registerACL(server) {
+    return new Promise(function (resolve, reject) {
+      require('./acl.js')(server, function(err) {
+        if (err) {
```
Comment (dmitrij-borchuk) [inline backend/server.js]: Use "err ? reject() : resolve();"

**Review by dmitrij-borchuk — CHANGES_REQUESTED** [review]

**Comment by dmitrij-borchuk** [comment]
Folder "dist" should be ignored

File: backend/dataConnection.js [hunk 13]
```diff
@@ -1,11 +1,11 @@
-var dbConnect = function(mysql) {
-  var connection = mysql.createConnection({
-    host: 'localhost',
-    user: 'root',
-    password: '',
-    database: db.dbName
-  });
+var mysql = require('mysql');
+var connection = mysql.createConnection({
+  host: 'localhost',
```
Comment (dmitrij-borchuk) [inline backend/dataConnection.js]: Move this settings to config

File: backend/dataConnection.js [hunk 14]
```diff
@@ -1,11 +1,11 @@
...
+  user: 'root',
+  password: ''
+});
 
+module.exports = function() {
```
Comment (dmitrij-borchuk) [inline backend/dataConnection.js:11]: It should return promise

File: backend/dataConnection.js [hunk 15]
```diff
@@ -1,11 +1,11 @@
...
   connection.connect(function(err){
     if(!err) {
         console.log("Database is connected ... ");
```
Comment (dmitrij-borchuk) [inline backend/dataConnection.js]: Remove console.log
It should call "promice.reject()"

**Review by dmitrij-borchuk — CHANGES_REQUESTED** [review]

File: backend/server.js [hunk 16]
```diff
@@ -23,11 +23,16 @@ module.exports = function () {
...
+    registerACL(server).then(
+      registerStaticFilesServer(server)
+    ).then(
       _.bind(registerRouting, null, server)
     ).then(
       _.bind(run, null, server)
+    // ).then(
+    //   _.bind(registerAuth, null, server)
+    ).then(
+      connectDB()
```
Comment (dmitrij-borchuk) [inline backend/server.js]: Should return result

**Comment by dmitrij-borchuk** [comment]
Remove forler "dist" from pull request

**Review by dmitrij-borchuk — CHANGES_REQUESTED** [review]

File: backend/dataConnection.js [hunk 17]
```diff
@@ -0,0 +1,25 @@
...
+module.exports = function() {
+  return new Promise(
+    function(resolve, reject) {
+
+      connection.connect(function(err){
+        err ? reject() : resolve();    
+      });
+
+      connection.query('SELECT * FROM pvmDB', function(err, rows, fields) {
+        err ? reject() : resolve();
```
Comment (dmitrij-borchuk) [inline backend/dataConnection.js]: "reject" should be calld with "arr" as a param

**Review by dmitrij-borchuk — CHANGES_REQUESTED** [review]

File: backend/dataConnection.js [hunk 18]
```diff
@@ -0,0 +1,25 @@
...
+  user: config.db.user,
+  password: config.db.password
+});
+
+module.exports = function() {
+  return new Promise(
+    function(resolve, reject) {
+
+      connection.connect(function(err){
+        err ? reject() : resolve();
```
Comment (dmitrij-borchuk) [inline backend/dataConnection.js]: "reject" should be calld with "arr" as a param

**Review by dmitrij-borchuk — CHANGES_REQUESTED** [review]

**Review by dmitrij-borchuk — APPROVED** [review]

**Pull request merged**